webapp/src/js/app.js

```diff
@@ -146,7 +146,7 @@ const createReduxLoggerConfig = Selectors => ({
     $compileProvider.aHrefSanitizationWhitelist(
       /^\s*(https?|ftp|mailto|tel|sms|file|blob):/
     );
-    var isDevelopment = window.location.hostname === 'localhost';
+    var isDevelopment = window.location.hostname === 'localhost' && window.location.port !== '9876'; // 9876 is port used for karma tests
```

Comment:
Currently a lot of our karma tests for controllers mock the redux actions/reducers. The tests for the tasks controller mock the redux store itself and confirm the data that is written to the store, not just the data that is passed to an action. It is a bit more of an integration test. I think this is a pattern we should encourage, particularly as we move more and more logic into redux. The redux logging that results for this style of test is quite verbose. It isn't a huge problem now, so happy to remove this line - but that was my thinking.

Reply:
I think we should keep a clear distinction between unit tests and integration tests, and therefore it would be more correct to mock the redux actions, and then write another unit test for redux which ensures the data is written to the store when the action is called.

As well as that you may want to write integration tests, but they serve a different purpose, and I don't want to confuse the issue by having some unit tests actually test _some_ integrations because it makes it unclear what our actual test coverage is.

The integration tests I really care about are end-to-end ones. For example, you're testing that the store is updated but not that the UI is updated when the store is written to. This would be relatively easy to write and test the integration from start to finish.

Reply:
although it is relatively easy to write a test that validates controller down to the UI, i find them cumbersome to maintain. they require xpaths to locations in the dom, which make them quite fragile.

> I think we should keep a clear distinction between unit tests and integration tests
where should i put integration tests then?

Reply:
>  i'd write that in protractor instead of karma.

Definitely write e2e tests in protractor as part of the e2e suites we already have.

> where should i put integration tests?

In API we have a `tests/integration` directory. We could introduce the same thing in webapp?